--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -78,7 +78,7 @@
     }
 
     kotlinOptions {
-        freeCompilerArgs = freeCompilerArgs + "-Xcontext-receivers"
+        freeCompilerArgs = freeCompilerArgs + "-Xcontext-receivers" + "-Xuse-k2-kapt"
         jvmTarget = "21"
     }
 
@@ -102,6 +102,17 @@
     arg("room.schemaLocation", "$projectDir/schemas")
 }
 
+kapt {
+    correctErrorTypes = true
+    arguments {
+        arg("dagger.fastInit", "ENABLED")
+    }
+    javacOptions {
+        option("-source", "21")
+        option("-target", "21")
+    }
+}
+
 dependencies {
     implementation(libs.guava)
     implementation(libs.coroutines.guava)
